### ISSUE TITLE
Option to assign filenames for individual files (.tsv etc.) in FragPipe workflow

### ISSUE THREAD

**Issue by tobimaile** [Nesvilab/FragPipe#980]
Hi everyone!

Is it possible to add an option (checkbox) in the Workflow tab that enables Fragpipe to assign filenames for the various individual files (peptide.tsv, ion.tsv, psm.tsv etc.) where the filename is drawn from the entry in the "Experiment" column? Possible with a separator that is machine-readable such as underscore or similar?

Example:
Experiment1_psm.tsv

I believe this would be super helpful for manual or machine-searched reading and uploading of the various subfiles in the various output folders without getting confused over filenames being identical.

**Comment by fcyu** [comment]
Thanks for the suggestion.

It should be helpful for some users, but it will also break some of the tools, such as IonQuant, TMT-Integrator, EasyPQP, because they assume the filename is without the experimental name. 

We need to discuss this internally but I would vote not to change it. It requires lots of effort but not much gain. BTW, MaxQuant output files also have the same name from different jobs so that downstream tools don't need to worry about the file name changes.

Best,

Fengchao

**Comment by tobimaile** [comment]
I understand! I believe my position is a bit unique compared to the average user as I am working more in the method development area where you regularly comb through ion and psm data rather than just the finished protein lists and ratios.

Maybe to avoid downstream implications, a copy function into a separate folder or a placeholder option in the code (e.g. *_psm.tsv) could work without creating too much hassle downstream and in the base code?

**Comment by fcyu** [comment]
> Maybe to avoid downstream implications, a copy function into a separate folder or a placeholder option in the code (e.g. *_psm.tsv) could work without creating too much hassle downstream and in the base code?

Yes, this could be a good solution. And this step does not have to be done by FragPipe. Any downstream program can do that at the beginning of the analysis.

Best,

Fengchao

**Comment by anesvi** [comment]
I suppose we can ask Felipe to add a tag to philosopher report command to write files as foldername_psm.tsv

**Comment by fcyu** [comment]
BTW, may I ask why you don't use the `combined_*.tsv` files? In this way, this issue will no longer exist.

Thanks,

Fengchao

**Comment by tobimaile** [comment]
I should have focused in the issue description on the files that I work most closely with at the moment, sorry for the confusion, here is some user feedback on what I do: 
I do use combined_*.tsv files extensively actually where picking out specific sample groups via code is no problem, so there is no immediate need for individual filenames for these. 
I currently investigate PSMs more closely and some variables in these files are very useful. Large studies with many runs create extremely large tables though, which I think is your reason for not printing a combined_PSM.tsv file. 
I'll create these via data wrangling with R (and I am not an expert) from the separate files, but often I and others have to sort through specific runs. Since variable coding languages besides R are used in my environment, it would be less hassle to have an immediate Fragpipe output data folder with all the separate PSM files to cherry-pick. 
Admittedly, this is an convenience issue of sorts for myself but it can come in handy for some applications in shared raw data depositories when you want to automatically load specific files with using filenames as a selector mechanism. To some lesser degree, this applies to the other individual.tsv files as well, especially if you want to merge data from the various files into one megadata set.
With individual data names, your risk of overwriting data due to identical names is much lower, thus improving/aiding computational quality control, especially for large study files.

**Comment by fcyu** [comment]
Thanks for the detailed explanation. 

Let us discuss figuring out an approach to generate `experiment_*.tsv` files and not break the downstream tools. Maybe adding an additional option, like Alexey suggested, to generate another set of files is a good idea.

Best,

Fengchao

**Comment by fcyu** [comment]
And I think it may be better to let FragPipe copy and rename the existing `*.tsv` files, not Philosopher. Because in this way, everything in the, for example, `psm.tsv` files will be copied. Some of the content in the `psm.tsv` files are updated/appended by downstream tools after running Philosopher.

Best,

Fengchao

**Comment by tobimaile** [comment]
Sounds great, as always, thank you very much for the rapid response!
